mailpoet/lib/API/JSON/v1/Setup.php

```diff
@@ -18,17 +19,23 @@ class Setup extends APIEndpoint {
   /** @var Activator */
   private $activator;
 
+  /** @var SettingsController */
+  private $settings;
+
   public function __construct(
     WPFunctions $wp,
-    Activator $activator
+    Activator $activator,
+    SettingsController $settings
   ) {
     $this->wp = $wp;
     $this->activator = $activator;
+    $this->settings = $settings;
   }
 
   public function reset() {
     try {
       $this->activator->deactivate();
+      $this->settings->resetCache();
```

Comment:
I'm still naive to the codebase, and I don't quite understand the motive behind this change. Could you please provide some context? A commit description would be even better for puzzled readers :wink:

Reply:
@mhadidg It's just that `$this->activator->deactivate()` wipes out the database tables, but since the following `$this->activator->activate()` is within the same request, the settings are not wiped out — it contains the old settings in memory. Do you think I should also amend the commit message?

Reply:
Aha, I get it. It is needed after the changes [here](https://github.com/mailpoet/mailpoet/pull/5818/commits/1a6aa487aae9f26081575caa63f3a0001f44166a)!

> Do you think I should also amend the commit message?

Not really, it's more like nice-to-have.